test/framework/toolchain.py

```diff
@@ -755,7 +755,7 @@ def test_old_new_iccifort(self):
         self.modtool.purge()
 
         tc = self.get_toolchain('goolfc', version='1.3.12')
-        tc.options.update({'openmp': True})
+        tc.set_options({'openmp': True})
```

Comment:
Just merge this PR, then #1966 will be updated, and should be good to go (and so will your PR for the ieee toolchain option, two birds with one stone!)